audit/record.go

```diff
@@ -33,5 +38,22 @@ func (rec *Record) AddMeta(name string, val interface{}) {
 	if rec.Meta == nil {
 		rec.Meta = Meta{}
 	}
+
+	// possibly convert val to something better suited for serializing
+	// via zero or more conversion functions.
+	var converted bool
+	for _, conv := range rec.metaConv {
+		val, converted = conv(val)
+		if converted {
+			break
+		}
+	}
```

Comment:
a simple sanity test that covers this block would be nice too

Reply:
I added unit tests for this via https://github.com/mattermost/mattermost-server/pull/14144/commits/fc8d74fb6c1d38fbfd63a14f5145d9451f0e36a2